app/models/manageiq/providers/terraform_enterprise/automation_manager/orchestration_stack/status.rb

```diff
@@ -0,0 +1,52 @@
+class ManageIQ::Providers::TerraformEnterprise::AutomationManager::OrchestrationStack::Status < ::OrchestrationStack::Status
+  def initialize(status)
+    self.status = status
+    self.reason = self.class.run_state_to_description(status)
+  end
+
+  def succeeded?
+    status.downcase == "planned_and_finished"
+  end
+
+  def failed?
+    %w[policy_soft_failed errored].include?(status.downcase)
+  end
+
+  def canceled?
+    %w[discarded canceled force_canceled].include?(status.downcase)
+  end
+
+  # Lookup table of run state descriptions
+  # https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#run-states
+  def self.run_state_to_description(status)
+    case status
```

Comment:
You did status.downcase in the other predicate methods, but not here. Do you need that here too? Does it make more sense to downcase it immediately on setting it in initialize?

Reply:
Yeah would be better to just do it in initialize, will push shortly